File: java/javaCoreLearning/src/closure/ClosureTest.java
package closure;

public class ClosureTest {
    public static void main(String[] args) {
        // java 中是没有函数的概念的, 这里用了
        // java 库中的 Runnable 接口。
        Runnable[] functionList = new Runnable[5];

        // 给上面的数组添加 Runnable 元素
        for (int i = 0; i < 5; i++) {
            int j = i;
            functionList[i] = () -> System.out.println(j);
        }

        // 依次执行数组中的 Runnable 元素
        for (Runnable elem : functionList) {
            elem.run();
        }
    }
}

File: java/javaCoreLearning/src/closure/ClosureTest.js
function closureTest() {
    /**
     * 首先来看当 identifier 是 javaScript 最开始就有的 var时
     */
    // 定义个数组用来储存指向函数的指针（理解成引用也可）
    var functionList =[];

    // 开始给上面的数组中添加元素
    for (var i = 0; i < 5; i++) {
        functionList.push(() => {
            console.log(i);
        })
    }

    // 依次执行上面数组中所储存的函数
    functionList.forEach((elem) => elem());
}

closureTest();